guildwars2/guild/general.py

```diff
@@ -15,25 +15,34 @@ class GeneralGuild:
         if ctx.invoked_subcommand is None:
             await self.bot.send_cmd_help(ctx)
 
-    @guild.command(name="info")
+    @guild.command(name="info", usage="<guild name>")
     @commands.cooldown(1, 20, BucketType.user)
-    async def guild_info(self, ctx, *, guild_name: str):
+    async def guild_info(self, ctx, *, guild_name=None):
         """General guild stats
 
         Required permissions: guilds
         """
+        # Read preferred guild from DB
         try:
-            endpoint_id = "guild/search?name=" + guild_name.replace(' ', '%20')
-            guild_id = await self.call_api(endpoint_id)
-            guild_id = guild_id[0]
-            endpoint = "guild/{0}".format(guild_id)
-            results = await self.call_api(endpoint, ctx.author, ["guilds"])
+            guild = await self.get_guild(ctx, guild_name=guild_name)
         except (IndexError, APINotFound):
             return await ctx.send("Invalid guild name")
         except APIError as e:
             return await self.error_handler(ctx, e)
+
+        if not guild:
+            return await self.bot.send_cmd_help(ctx)
```

Comment:
Found a more elegant way to do this:
at the top of the file add
```py
from discord.ext.commands.errors import BadArgument
```
now here you can do:
```py
if not guild:
    raise BadArgument
```
This will send the cmd help, and more importantly, reset the command cooldown - which are quite lengthy. 
Same can be done for other send_cmd_help's in other `guild` commands

Reply:
Replaced send_cmd_help's in guild commands with BadArgument